patterns/distributed/dev/group_vars/apim.yml

```diff
@@ -0,0 +1,173 @@
+
+# Set the location the product packages should reside in (eg: "local" in the /files directory, "remote" in a remote location)
+pack_location: local
+#pack_location: remote
+#remote_jdk: "<URL_TO_JDK_FILE>"
+#remote_pack: "<URL_TO_APIM_PACK>"
+
+# JDK Distributions
+setup_java_enabled: true #If this is set to false java_home variable below needs to be specified.
+jdk_name: amazon-corretto-17.0.6.10.1-linux-x64 # Update this as per jdk version you use
+java_dir: /opt
+java_symlink: "{{ java_dir }}/java"
+java_home: "{{ java_dir }}/{{ jdk_name }}"
+
+# Server URL of the Authentication service. Make sure to import the Key Manager's public certificate to WSO2 API-M's
+# client-truststore.jks. For more information, see https://docs.wso2.com/display/ADMIN44x/Creating+New+Keystores
+key_manager_server_url: https://{{ cp_hostname }}:9443/services/
+
+# API Store related configurations
+api_devportal_url: https://{{ cp_hostname }}:9443/devportal
+api_devportal_server_url: https://{{ cp_hostname }}:9443/services/
+
+# Gateway configurations
+gateway_environments:
+  - { type: 'hybrid',
+      name: 'Default',
+      description: 'This is a hybrid gateway that handles both production and sandbox token traffic.',
+      service_url: 'https://{{ gw_hostname }}:9443/services/',
+      ws_endpoint: 'ws://{{ gw_hostname }}:9099',
+      wss_endpoint: 'wss://{{ gw_hostname }}:8099',
+      http_endpoint: 'http://{{ gw_hostname }}:8280',
+      https_endpoint: 'https://{{ gw_hostname }}:8243',
+      websub_event_receiver_http_endpoint: "http://{{ gw_hostname }}:9021",
+      websub_event_receiver_https_endpoint: "https://{{ gw_hostname }}:8021"}
+
+sync_runtime_gateway_labels: ["Default"]
+
+# Traffic Manager configurations
+traffic_manager_host_url: https://{{ tm_hostname }}:9443
+traffic_manager_receiver_url: tcp://{{ tm_hostname }}:${receiver.url.port}
+traffic_manager_auth_url: ssl://{{ tm_hostname }}:${auth.url.port}
+throttle_config_policy_deployer_url: https://{{ tm_hostname }}:9443/services/
+notification_endpoint_url: https://{{ tm_hostname }}:9443
+throttle_decision_endpoints:
+  - tcp://{{ tm_hostname }}:5672
+throttling_url_group:
+  - traffic_manager_urls:
+      - tcp://{{ tm_hostname }}:9611
+    traffic_manager_auth_urls:
+      - ssl://{{ tm_hostname }}:9711
+
+# Workflow configurations
+workflow_enable: "false"
+workflow_service_url: https://{{ cp_hostname }}:9445/bpmn
+workflow_callback_endpoint: https://{{ cp_hostname }}:9443/api/am/publisher/v0.17/workflows/update-workflow-status
+workflow_token_endpoint: https://{{ cp_hostname }}:9443/oauth2/token
+workflow_client_registration_endpoint: https://{{ cp_hostname }}:9443/client-registration/v0.17/register
+workflow_server_url: https://apis.apim.com:9443/services/  #Developer Portal Service URL
+
+# Datasource configurations
+wso2shared_db_type: mysql
+wso2shared_db_url: jdbc:mysql://localhost:3306/shared_db?ssl-mode=REQUIRED
```

Comment:
May need to add a variable called "db_hostname" and replace localhost in line 134 and 140

Reply:
This was kept as it is since other information also may need to change according to the user's DB configurations. For example DB type, port, database names. So my opinion is giving the option to change the entire DB URL would be make the configurations much simpler.

Reply:
Agreed